app/services/notification/ScheduledNotificationRunner.scala

```diff
@@ -1,22 +1,33 @@
 package services.notification
 
+import javax.inject.{Inject, Singleton}
+
 import com.amazonaws.services.simpleemail.model._
-import config.AMIableConfig
+import config.{AMIableConfig, AmiableConfigProvider}
 import models._
+import play.api.{Environment, Logger, Mode}
 import prism.{Prism, PrismLogic}
 
 import scala.concurrent.ExecutionContext
 
-object ScheduledNotificationRunner {
-  def run(mailClient: AWSMailClient)(implicit config: AMIableConfig, ec: ExecutionContext): Attempt[List[String]] = {
+@Singleton
+class ScheduledNotificationRunner@Inject()(mailClient: AWSMailClient, environment: Environment, amiableConfigProvider: AmiableConfigProvider)(implicit ec: ExecutionContext) {
+  implicit val config = amiableConfigProvider.conf
+
+  def run(): Attempt[List[String]] = {
     for {
       instancesWithAmis <- Prism.instancesWithAmis(SSA(stage = Some("PROD")))
       oldInstances = PrismLogic.oldInstances(instancesWithAmis)
       ownersWithDefault <- Prism.getOwners
       ownersAndOldInstances = findInstanceOwners(oldInstances, ownersWithDefault)
       mailIds <- Attempt.traverse(ownersAndOldInstances.toList) { case (owner, oldInstancesForOwner) =>
-        val request = createEmailRequest(owner, oldInstancesForOwner)
-        mailClient.send(owner, request)
+        val instances = oldInstancesForOwner.sortBy(i => (i.stack, i.stage, i.app.headOption))
+        val request = createEmailRequest(owner, instances, config)
+        if(environment.mode == Mode.Prod || config.overrideToAddress.nonEmpty)
```

Comment:
Thought to leave the logic here as otherwise we would need to have a global override address which might be confusing in other use cases